utilities/rocAL/rocAL_unittests/pixel_comparison/image_comparison.py

```diff
@@ -0,0 +1,148 @@
+        else:
+            golden_file_path = aug_name
+
+        #For randomized augmentation
+        if(file_name_s[0] in randomized_augmentation):
+            total_case_count=total_case_count+1
+            augmentation_name = aug_name.split('.')[0]
+            logging.info("Running %s",augmentation_name)
+            passed_case_count=passed_case_count+1
+            logging.info("PASSED ")
+        elif golden_file_path in golden_output_dir_list:
+            total_case_count=total_case_count+1
+            ref_file_path = ref_output_path+golden_file_path
+            rocal_file_path = rocal_output_path+aug_name
+            if(os.path.exists(rocal_file_path) and os.path.exists(ref_file_path)):
+                logging.info("Running %s ",aug_name.split('.')[0])
+                img1 = Image.open(ref_file_path)
+                img2 = Image.open(rocal_file_path)
+
+                # Check if the images have the same dimensions
+                if img1.size != img2.size:
+                    logging.info("The images have different sizes.")
+                    exit()
+
+                # Compare the pixel values for each image
+                pixeldiff = None
+                tot_count = 0
+                if 'larger' in aug_name:
+                    resize_width = 400
+                    resize_height = 300
+                    image_offset = 400
+                    pixeldiff, tot_count = compare_pixels(img1, img2, aug_name, resize_width, resize_height)
+                    pixeldiff2, tot_count2 = compare_pixels(img1, img2, aug_name, resize_width, resize_height, image_offset)
+                    pixeldiff = [x + y for x, y in zip(pixeldiff, pixeldiff2)]
+                    tot_count = tot_count + tot_count2
+                elif 'smaller' in aug_name:
+                    resize_width = 533
+                    resize_height = 400
+                    image_offset = 2400
+                    pixeldiff, tot_count = compare_pixels(img1, img2, aug_name, resize_width, resize_height)
+                    pixeldiff2, tot_count2 = compare_pixels(img1, img2, aug_name, resize_width, resize_height, image_offset)
+                    pixeldiff = [x + y for x, y in zip(pixeldiff, pixeldiff2)]
+                    tot_count = tot_count + tot_count2
+                else:
+                    pixeldiff, tot_count = compare_pixels(img1, img2, aug_name, img1.size[0], img1.size[1]) 
+                total_pixel_diff = 0
+                for pix_diff in range(1,6):
+                    total_pixel_diff += pixeldiff[pix_diff]
+                if (total_pixel_diff):
+                    failed_case_list.append(golden_file_path)
+                    failed_case_count=failed_case_count+1
+                    logging.info("FAILED")
+                    logging.info("Printing pixel mismatch %s",pixeldiff)
+                    mismatch_percentage = round((total_pixel_diff/tot_count)*100,2)
+                    if(mismatch_percentage):
```

Comment:
@fiona-gladwin why are we printing only for some conditions? This avoids printing % mismatch for pixel wise for many cases. Please let it print for all cases and remove this if condition